arches/app/utils/frontend_configuration_utils.py

```diff
@@ -0,0 +1,198 @@
+import json
+import os
+import re
+import site
+import sys
+
+from django.conf import settings
+from django.urls import get_resolver, URLPattern, URLResolver
+from django.urls.resolvers import RegexPattern, RoutePattern, LocalePrefixPattern
+
+from arches.settings_utils import list_arches_app_names, list_arches_app_paths
+
+
+def generate_frontend_configuration():
+    try:
+        _generate_frontend_configuration_directory()
+        _generate_urls_json()
+        _generate_webpack_configuration()
+        _generate_tsconfig_paths()
+    except Exception as e:
+        # Ensures error message is shown if error encountered
+        sys.stderr.write(str(e))
+        raise e
+
+
+def _generate_frontend_configuration_directory():
+    destination_dir = os.path.realpath(
+        os.path.join(_get_base_path(), "..", "frontend_configuration")
+    )
+
+    os.makedirs(destination_dir, exist_ok=True)
+
+
+def _generate_urls_json():
+    def generate_human_readable_urls(patterns, prefix="", namespace="", result={}):
```

Comment:
I try to avoid mutable default arguments like `[]` and `{}`, although I do see you've wrapped this all in a closure. Just wondering if you'd prefer to make this a little more explicit so someone else isn't surprised that `result` persists between calls.